doctr/models/detection/linknet/tensorflow.py

```diff
@@ -154,17 +159,30 @@ def compute_loss(
 
         seg_target = tf.convert_to_tensor(seg_target, dtype=out_map.dtype)
         seg_mask = tf.convert_to_tensor(seg_mask, dtype=tf.bool)
+        seg_mask = tf.cast(seg_mask, tf.float32)
 
-        # BCE loss
         bce_loss = tf.keras.losses.binary_crossentropy(seg_target, out_map, from_logits=True)[..., None]
+        proba_map = tf.sigmoid(out_map)
+
+        # Focal loss
+        if gamma < 0:
+            raise ValueError("Value of gamma should be greater than or equal to zero.")
+        # Convert logits to prob, compute gamma factor
+        p_t = (seg_target * proba_map) + ((1 - seg_target) * (1 - proba_map))
+        modulating_factor = tf.pow((1.0 - p_t), gamma)
+        # Compute alpha factor
+        alpha_factor = seg_target * alpha + (1 - seg_target) * (1 - alpha)
+        # compute the final loss
+        focal_loss = tf.reduce_sum(
+            seg_mask * alpha_factor * modulating_factor * bce_loss, (0, 1, 2)
+        ) / tf.reduce_sum(seg_mask, (0, 1, 2))
```

Comment:
So that we keep the same structure with the PyTorch implementation and stick with paper notations
```python
alpha = seg_target * alpha + (1 - seg_target) * (1 - alpha)
# Unreduced loss
focal_loss = alpha * (1 - p_t) ** gamma * bce_loss
# Class reduced
focal_loss = tf.reduce_sum(seg_mask * focal_loss, (0, 1, 2)) / tf.reduce_sum(seg_mask, (0, 1, 2))
```